src/core/mesh/qgsmeshlayerrenderer.h

```diff
@@ -74,7 +75,7 @@ struct CORE_NO_EXPORT QgsMeshLayerRendererCache
   double mVectorDatasetGroupMagMinimum = std::numeric_limits<double>::quiet_NaN();
   double mVectorDatasetGroupMagMaximum = std::numeric_limits<double>::quiet_NaN();
   bool mVectorDataOnVertices = true;
-
+  std::unique_ptr<QgsMesh3dAveragingMethod> mVectorAveragingMethod;
```

Comment:
do we need a separate scalar/vector avg method? given that it's a global layer-level setting, that should not be needed...

Reply:
i would rather keep it this way, it would complicate the logic of creation of the cache quite a lot to not much readable code...